indra/newview/llviewerparcelmgr.cpp

```diff
@@ -1332,7 +1332,7 @@ void LLViewerParcelMgr::sendParcelPropertiesUpdate(LLParcel* parcel, bool use_ag
     if(!parcel)
         return;
 
-    LLViewerRegion *region = use_agent_region ? gAgent.getRegion() : LLWorld::getInstance()->getRegionFromPosGlobal( mWestSouth );
+    LLViewerRegion *region = LLWorld::getInstance()->getRegionFromID(parcel->getRegionID());
```

Comment:
This is where the bug lived.  The region pointer was being found based on (a) whether the agent was in the region or (b) a cached value of `mWestSouth` which was supposed to be the global-position of the region's origin corner, but could sometimes be from the wrong region.

The fix is to cache the region_id in each parcel so it can be used to find the region in this context.